src/qibolab/_core/instruments/qblox/config.py

```diff
@@ -0,0 +1,153 @@
+from qibolab._core.serialize import Model
+
+from .identifiers import SlotId
+from .sequence import Q1Sequence
+
+__all__ = []
+
+
+class PortAddress(Model):
+    slot: SlotId
+    ports: tuple[int, Optional[int]]
+    input: bool = False
+
+    @classmethod
+    def from_path(cls, path: str):
+        """Load address from :attr:`qibolab.Channel.path`."""
+        els = path.split("/")
+        assert len(els) == 2
+        ports = els[1][1:].split("_")
+        assert 1 <= len(ports) <= 2
+        return cls(
+            slot=int(els[0]),
+            ports=(int(ports[0]), int(ports[1]) if len(ports) == 2 else None),
+            input=els[1][0] == "i",
+        )
+
+    @property
+    def local_address(self):
+        """Physical address within the module.
+
+        It will generate a string in the format ``<direction><channel>`` or
+        ``<direction><I-channel>_<Q-channel>``.
+        ``<direction>`` is ``in`` for a connection between an input and the acquisition
+        path, ``out`` for a connection from the waveform generator to an output, or
+        ``io`` to do both.
+        The channels must be integer channel indices.
+        Only one channel is present for a real mode operating sequencer; two channels
+        are used for complex mode.
+
+        .. note::
+
+            Description adapted from
+            https://docs.qblox.com/en/main/api_reference/cluster.html#qblox_instruments.Cluster.connect_sequencer
+        """
+        direction = "in" if self.input else "out"
+        channels = (
+            str(self.ports[0])
+            if self.ports[1] is None
+            else f"{self.ports[0]}_{self.ports[1]}"
+        )
+        return f"{direction}{channels}"
+
+
+def _probe(id_: ChannelId, channel: Channel) -> Optional[ChannelId]:
+    return (
+        id_
+        if isinstance(channel, IqChannel)
+        else (
+            channel.probe
+            if isinstance(channel, AcquisitionChannel) and channel.probe is not None
+            else None
+        )
+    )
+
+
+def module(
+    mod: Module,
+    mod_channels: dict[ChannelId, PortAddress],
+    channels: dict[ChannelId, Channel],
+    configs: Configs,
+):
+    # map sequencers to specific outputs (but first disable all sequencer connections)
+    mod.disconnect_outputs()
+
+    if mod.is_qrm_type:
+        # we do not currently support acquisition on external digital trigger
+        mod.scope_acq_trigger_mode_path0("sequencer")
+        mod.scope_acq_trigger_mode_path1("sequencer")
+
+    # set lo frequencies
+    los = {
+        (probe, cast(IqChannel, channels[probe]).lo)
+        for probe in (_probe(ch, channels[ch]) for ch in mod_channels)
+        if probe is not None
+    }
+    for probe, lo in los:
+        if lo is None:
+            continue
+        n = mod_channels[probe].ports[0]  # TODO: check it is the correct path
+        getattr(mod, f"out{n}_lo_freq")(cast(OscillatorConfig, configs[lo]).frequency)
```

Comment:
Checking a bit more carefully, the solution may be
```suggestion
        n = mod_channels[probe].ports[0] - 1
        getattr(mod, f"out{n}_lo_freq")(cast(OscillatorConfig, configs[lo]).frequency)
```
This fixes the error I am observing but I am not 100% sure it is a generally correct solution. I think the port numbers start from 1, but the LO outs from 0.

Reply:
> This fixes the error I am observing but I am not 100% sure it is a generally correct solution. I think the port numbers start from 1, but the LO outs from 0.

Yes, indeed.

This confusion may be generated by Qblox itself
- some API uses `out0` to `out3` https://docs.qblox.com/en/main/api_reference/module.html#QCM.out0_offset
- the module itself is calling them `O1` to `O4`, and I believe this is also what you need to specify to connect the sequencer, but I may be wrong https://docs.qblox.com/en/main/cluster/qcm.html

Let's assume this fix to be working, but leave the comment open. If confirmed, I will ask further explanation (or just provide some feedback) to the Qblox people.